analyzers/golang/analyze.go

```diff
@@ -107,79 +107,94 @@ func (a *Analyzer) Analyze() (graph.Deps, error) {
 
 	log.Debugf("Resolver: %#v", r)
 
-	// Use `go list` to get imports and deps of module.
-	main, err := a.Go.ListOne(m.BuildTarget)
-	if err != nil {
-		return graph.Deps{}, err
-	}
-	log.Debugf("Go main package: %#v", main)
-	deps, err := a.Go.List(main.Deps)
-	if err != nil {
-		return graph.Deps{}, err
-	}
-
-	// Construct map of import path to package.
-	gopkgs := append(deps, main)
-	gopkgMap := make(map[string]gocmd.Package)
-	for _, p := range gopkgs {
-		gopkgMap[p.ImportPath] = p
-	}
-	// cgo imports don't have revisions.
-	gopkgMap["C"] = gocmd.Package{
-		Name:     "C",
-		IsStdLib: true, // This is so we don't try to lookup a revision. Maybe there should be a NoRevision bool field?
-	}
-	log.Debugf("gopkgMap: %#v", gopkgMap)
-
-	// Construct transitive dependency graph.
-	pkgs := make(map[pkg.ID]pkg.Package)
-	for _, gopkg := range deps {
-		log.Debugf("Getting revision for: %#v", gopkg)
-
-		// Resolve dependency.
-		revision, err := a.Revision(project, r, gopkg)
+	var allImports []pkg.Import
+	importMap := make(map[pkg.Import]bool)
+	transitiveDeps := make(map[pkg.ID]pkg.Package)
+
+	for _, buildTag := range a.BuildTags {
```

Comment:
[process] This section looks like it's literally what we had before but put into a for-loop. It would be really helpful if you added a GH comment here before review indicating that. In general, it would be really helpful having a description of the high-level approach e.g.:

- Pass flags to the Go buildtool
- Run analysis in a for loop with all build tags
- Add fixtures for testing build tags
- Minor style nits on other files

Reply:
Good point, I updated the top comment to clarify for anyone who observes the PR in the future.